### PR TITLE
[exporter/datadog]: improve mappings for span kind dd span type

### PR DESCRIPTION
**Pull request by ericmustin**
**Description:** <Describe what has changed.>

This PR Adds some additional mappings between opentelemetry span kind and datadog span type, that allow users to see more specialized UIs and representation of traces properly marked `db` or `cache` whereas previously they were all `custom`

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.> updated unit tests

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by mx-psi — CHANGES_REQUESTED** [review]

File: exporter/datadogexporter/translate_traces.go [hunk 1]
```diff
@@ -43,9 +43,13 @@ const (
...
 	okCode              int32  = 0
-	httpKind            string = "http"
-	webKind             string = "web"
-	customKind          string = "custom"
+	kindDb              string = "db"
+	kindHTTP            string = "http"
+	kindWeb             string = "web"
+	kindCustom          string = "custom"
+	kindCache           string = "cache"
+	kindMemcached       string = "memcached"
```
[the remark this answers is not in the document]
Comment (ericmustin) [reply exporter/datadogexporter/translate_traces.go:51]: sounds good, made a pr here https://github.com/open-telemetry/opentelemetry-specification/pull/1689

File: exporter/datadogexporter/translate_traces.go [hunk 2]
```diff
@@ -376,16 +380,31 @@ func buildDatadogContainerTags(spanTags map[string]string) string {
...
+// db or cache spans, so we can reliably use it as a heuristic for infering the
+// difference between client, db, and cache span types. The only "cache" spans
+// in datadog currently are redis and memcached, so those are the only two db.system
+// attribute values we have to check to determine whether it's a db or cache span
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#semantic-conventions-for-database-client-calls
+func inferDatadogType(kind pdata.SpanKind, datadogTags map[string]string) string {
 	switch kind {
 	case pdata.SpanKindCLIENT:
-		return httpKind
+		if dbSysOtlp, okDbSysOtlp := datadogTags[conventions.AttributeDBSystem]; okDbSysOtlp {
```
[the remark this answers is not in the document]
Comment (ericmustin) [reply exporter/datadogexporter/translate_traces.go]: updated

File: exporter/datadogexporter/translate_traces_test.go [hunk 3]
```diff
@@ -1058,13 +1058,26 @@ func TestSpanNameNormalization(t *testing.T) {
...
+	}
+
+	spanTypeDb := inferDatadogType(pdata.SpanKindCLIENT, ddTagsDb)
+	spanTypeCache := inferDatadogType(pdata.SpanKindCLIENT, ddTagsCache)
 
 	assert.Equal(t, "http", spanTypeClient)
 	assert.Equal(t, "web", spanTypeServer)
 	assert.Equal(t, "custom", spanTypeCustom)
+	assert.Equal(t, "db", spanTypeDb)
+	assert.Equal(t, "cache", spanTypeCache)
```
[the remark this answers is not in the document]
Comment (ericmustin) [reply exporter/datadogexporter/translate_traces_test.go:1085]: updated

File: exporter/datadogexporter/translate_traces_test.go [hunk 4]
```diff
@@ -1058,13 +1058,26 @@ func TestSpanNameNormalization(t *testing.T) {
...
 func TestSpanTypeTranslation(t *testing.T) {
-	spanTypeClient := spanKindToDatadogType(pdata.SpanKindCLIENT)
-	spanTypeServer := spanKindToDatadogType(pdata.SpanKindSERVER)
-	spanTypeCustom := spanKindToDatadogType(pdata.SpanKindUNSPECIFIED)
+	spanTypeClient := inferDatadogType(pdata.SpanKindCLIENT, map[string]string{})
+	spanTypeServer := inferDatadogType(pdata.SpanKindSERVER, map[string]string{})
+	spanTypeCustom := inferDatadogType(pdata.SpanKindUNSPECIFIED, map[string]string{})
+
+	ddTagsDb := map[string]string{
+		"db.system": "postgres",
```
[the remark this answers is not in the document]
Comment (ericmustin) [reply exporter/datadogexporter/translate_traces_test.go]: updated

**Review by mx-psi — APPROVED** [review]

**Pull request merged**